unlock-protocol.com/blog/version-1-2.md

```diff
@@ -0,0 +1,87 @@
+
+## Unlock
+
+More importantly, you can interact with the Unlock smart contract itself. For example, you can view view [the owner of the Unlock contract](https://etherscan.io/address/0xa39b44c4affbb56b76a1bf1d19eb93a5dfc2eba9), the global token symbol for all NFT: `KEY` or even the current version of the Unlock contract (we use `5` internally and `1.2` externally).
+
+Another very important element that you can view is the _Gross Network Product_ which is the total sum of ethereum spent to purchase lock keys, as long as the key was priced in Eth, DAI, SAI or BAT. We use Uniswap to get a price feed when we need to convert to Ether, but that's for another post ;) Measuring the Gross Network Product is a critical piece of the [Unlock Discount Token](https://github.com/unlock-protocol/unlock/wiki/The-Unlock-Discount-Tokens) model that we are working on to incentivize membership referrals.
+
+By using Etherscan you can also deploy locks, without having to go through [our Dashboard](https://app.unlock-protocol.com/dashboard/), using the `createLock` method directly. The new version deployed on Monday uses the new Ethereum [`create2` method](https://eips.ethereum.org/EIPS/eip-1014) under the hood. This allows creators to deploy locks at only 15% of the cost that it took before that! In order to achieve this, we had to deploy a so [called `template` contract](https://etherscan.io/address/0x2d5487420fbeb5ba74eadf51084d4f71e1733983).
+
+
+# The new public locks
+
+First, I want to repeat that old the **locks deployed before Monday have stayed absolutely unchanged**. It is actually impossible to change them. Once deployed, their code cannot be altered. This is a very important part of the "decentralization promise" that we make to creators: nobody will be able to change the terms of your memberships, but you.
+
+Now, if you deploy a lock now, it will come with new features or improvements! Here are a few of the most exciting ones.
+
+## On Key Sold hook
+
+We strongly believe that one of the most important benefits of building on Ethereum is the **composability**. In practice, this is the ability for contracts to interact for the benefits of users. For example, purchasing a key using an ERC20 is an example of composability. Now, with hooks, the lock owner can define what other contracts will be able to run when a key has been sold.
+
+Why is this useful? Here are 2 of my favorite examples:
+
+1. _Matching donations_: someone could agree to match donations so that for any key purchase on a donation lock, they would make a payment to the lock as well. The "matching" contract could even query the lock itself to know how many keys have been sold and make a quadratic matching!
```

Comment:
On a related note - we might want to spend a little time creating a good sample contract using hooks.  This is a cool use case we could try.

Reply:
yes!